### PR TITLE
Return success on failure to decrypt AES success action

### PR DESCRIPTION
**Pull request by andrei-21**
If there is an AES success action in `payRequest` (LUD-10) then failure to decrypt ciphertext should not fail the payment, since the funds were actually sent and the lightning payment has succeeded.

Solves #632.

### CODE REVIEW

File: libs/sdk-core/src/lnurl/pay.rs [hunk 1]
```diff
@@ -236,7 +236,9 @@ pub(crate) mod model {
         /// See [SuccessAction::Aes] for received payload
         ///
         /// See [AesSuccessActionDataDecrypted] for decrypted payload
-        Aes { data: AesSuccessActionDataDecrypted },
+        Aes {
+            result: Result<AesSuccessActionDataDecrypted, String>,
```
[the remark this answers is not in the document]
Comment (ok300) [reply libs/sdk-core/src/lnurl/pay.rs:240]: `Result` is a good direction, try to use a`Result`-like enum with an inner field. These play better with bindings and uniffi AFAIK.

Something like

```rust
    pub enum AesSuccessActionDataResult {
        Decrypted { data: AesSuccessActionDataDecrypted },
        Error { data: String }
    }
```

To propagate the changes to the bindings for other languages, you'd have to
- update `breez_sdk.udl`
- run `cargo build` in `sdk-bindings`
- run `make init` / `make dev` in `sdk-flutter`
- run the command listed in the README of https://github.com/breez/breez-sdk-rn-generator

**Review by dangeross — COMMENTED** [review]
Overall looks good I think, just some comments around the error variant variables

**Review by ok300 — CHANGES_REQUESTED** [review]
Looks good!

Could you please also add a short test case for when decryption fails?

There is `test_lnurl_pay_aes_success_action` that covers the success scenario, I imagine a similar one can check the new result type in case of failed decryption.

**Review by ok300 — APPROVED** [review]
My bad, I missed it.

Then it's all good!

**Review by roeierez — APPROVED** [review]
LGTM

**Pull request merged**